solr/core/src/test/org/apache/solr/handler/admin/api/BackupCoreAPITest.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.core.backup.BackupFilePaths;
+import org.apache.solr.core.backup.repository.BackupRepository;
+import org.apache.solr.handler.IncrementalShardBackup;
+import org.apache.solr.handler.SnapShooter;
+import org.apache.solr.handler.admin.CoreAdminHandler;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BackupCoreAPITest extends SolrTestCaseJ4 {
+
+  private BackupCoreAPI backupCoreAPI;
+
+  @BeforeClass
+  public static void initializeCoreAndRequestFactory() throws Exception {
+    initCore("solrconfig.xml", "schema.xml");
+
+    lrf = h.getRequestFactory("/api", 0, 10);
+  }
+
+  @Before
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    SolrQueryRequest solrQueryRequest = req();
+    SolrQueryResponse solrQueryResponse = new SolrQueryResponse();
+    CoreContainer coreContainer = h.getCoreContainer();
+
+    CoreAdminHandler.CoreAdminAsyncTracker coreAdminAsyncTracker =
+        new CoreAdminHandler.CoreAdminAsyncTracker();
+    backupCoreAPI =
+        new BackupCoreAPI(
+            coreContainer, solrQueryRequest, solrQueryResponse, coreAdminAsyncTracker);
+  }
+
+  @Test
+  public void testCreateBackupReturnsValidResponse() throws Exception {
+    final String backupName = "my-new-backup";
+    BackupCoreAPI.BackupCoreRequestBody backupCoreRequestBody = createBackupCoreRequestBody();
+    backupCoreRequestBody.incremental = false;
+    SnapShooter.SnapShooterBackupCoreResponse response =
+        (SnapShooter.SnapShooterBackupCoreResponse)
+            backupCoreAPI.createBackup(coreName, backupName, backupCoreRequestBody, null);
+
+    assertEquals(backupName, response.snapshotName);
+    assertEquals("snapshot." + backupName, response.directoryName);
+    assertEquals(1, response.fileCount);
+    assertEquals(1, response.indexFileCount);
+  }
+
+  @Test
+  public void testMissingLocationParameter() throws Exception {
+    final String backupName = "my-new-backup";
+    BackupCoreAPI.BackupCoreRequestBody backupCoreRequestBody = createBackupCoreRequestBody();
+    backupCoreRequestBody.location = null;
+    backupCoreRequestBody.incremental = false;
+    final SolrException solrException =
+        expectThrows(
+            SolrException.class,
+            () -> {
+              backupCoreAPI.createBackup(coreName, backupName, backupCoreRequestBody, null);
+            });
+    assertEquals(500, solrException.code());
+    assertTrue(
+        "Exception message differed from expected: " + solrException.getMessage(),
+        solrException.getMessage().contains("'location' is not specified as a query"));
+  }
+
+  @Test
+  public void testMissingCoreNameParameter() throws Exception {
+    final String backupName = "my-new-backup";
```

Comment:
[0] If you wanted you could create a static constant at the top of the test, and then use that in each test method.